smtp/client.ts

```diff
@@ -63,12 +63,32 @@ export class SMTPClient {
 				}
 				this.queue.push(stack);
 				this._poll();
+				if (this.queue.every((x) => x !== stack)) {
```

Comment:
i'm a bit confused. is everything in the `queue` (before `stack`) a sent message?

Reply:
from my reading, the `queue` only contains _unsent_ messages; however it does seem like there's a window after a connection is established where a message can be de`queue`d and an error can still surface (e.g. from greylisting).

now that i write that out, i'm realizing i should add that as a test case. i'm guessing it'll break this assumption, in which case i'll have to either rethink the success check or rework `queue` behavior.

Reply:
what is the interpretation of this part of the logic then?

Reply:
currently, it is "if the message is not in the queue, it was successfully sent"

Reply:
turns out it isn't necessary to track if messages are sent, as the message's callback is always called deeper in the stack. this also means i've been wrong all this time & it's always been possible to `await` the results of a promisified send. neat!